hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -278,21 +292,28 @@
       name: chip_pattgen_ios
       desc: '''Verify pattern generation to chip output pads.
 
-            SW programs pattgen to generate distinct patterns on both groups. SW programs pinmux to
-            select pattgen outputs to be routed. SW validates the reception of patt_done interrupts.
-            Testbench verifies the correctness of the pattern seen on the IO pins.
+            - Program the pattgen to generate a known pattern in each lane.
+            - Program the pinmux to route the chosen output to the chip IOs.
+            - Verify that the correct pattern is seen on the IOs by hooking up the pattgen monitor.
```

Comment:
Is there an example yet as to how the scoreboards or monitors work at top-level?

To elaborate on the question: Is there any SV sequence generation at all?  My understanding so far is that these are all driving by a test C application.  That said, I'm really interested to learn more about how we are going to connect this application to the monitors on the backside, and how they interact with the driving application.

Reply:
Yes there already are few tests of this sort that have been developed and are running. Please see UART as an example:
https://github.com/lowRISC/opentitan/blob/9632c45bff7dc6f24f48af4daf2bb0ba8f79252c/hw/top_earlgrey/dv/chip_sim_cfg.hjson#L156

It pairs the [SW test](https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/sim_dv/uart_tx_rx_test.c) with an [SV sequence](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/dv/env/seq_lib/chip_sw_uart_tx_rx_vseq.sv). This SV sequence extends from the [SW base sequence](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/dv/env/seq_lib/chip_sw_base_vseq.sv), which initializes the memories with SW test images before the CPU is brought out of reset. 

The chip env [instantiates](https://github.com/lowRISC/opentitan/blob/9632c45bff7dc6f24f48af4daf2bb0ba8f79252c/hw/top_earlgrey/dv/env/chip_env.sv#L102) the UART agent, which monitors the TX and RX port on chip's IOs and collects and sends the data. The chip sequencer is a [subscriber](https://github.com/lowRISC/opentitan/blob/9632c45bff7dc6f24f48af4daf2bb0ba8f79252c/hw/top_earlgrey/dv/env/chip_env.sv#L134) of the UART monitor TLM channel so that the sequence can [perform the checks](https://github.com/lowRISC/opentitan/blob/9632c45bff7dc6f24f48af4daf2bb0ba8f79252c/hw/top_earlgrey/dv/env/seq_lib/chip_sw_uart_tx_rx_vseq.sv#L47) directly.  

We don't reuse the IP level environments or scoreboards at the chip level. We are not developing the chip level scoreboard either because it is a little complicated.